rajawali/src/main/java/org/rajawali3d/Object3D.java

```diff
@@ -174,30 +175,50 @@ protected void preRender() {
 	 * @param vpMatrix {@link Matrix4} The view-projection matrix
 	 * @param projMatrix {@link Matrix4} The projection matrix
 	 * @param vMatrix {@link Matrix4} The view matrix
-	 * @param pickerInfo The current color picker info. This is only used when an object is touched.
+	 * @param sceneMaterial The scene-wide Material to use, if any.
 	 */
 	public void render(Camera camera, final Matrix4 vpMatrix, final Matrix4 projMatrix,
 			final Matrix4 vMatrix, Material sceneMaterial) {
-		render(camera, vpMatrix, projMatrix, vMatrix, null, sceneMaterial);
-	}
+		render(camera, vpMatrix, projMatrix, vMatrix, null, sceneMaterial, false);
+	}
+
+    /**
+     * Renders the object with a parent matrix
+     *
+     * @param camera The camera
+     * @param vpMatrix {@link Matrix4} The view-projection matrix
+     * @param projMatrix {@link Matrix4} The projection matrix
+     * @param vMatrix {@link Matrix4} The view matrix
+     * @param parentMatrix {@link Matrix4} This object's parent matrix
+     * @param sceneMaterial The scene-wide Material to use, if any.
+     */
+    public void render(Camera camera, final Matrix4 vpMatrix, final Matrix4 projMatrix, final Matrix4 vMatrix,
+                       final Matrix4 parentMatrix, Material sceneMaterial) {
+        render(camera, vpMatrix, projMatrix, vMatrix, parentMatrix, sceneMaterial, false);
+    }
 
-	/**
-	 * Renders the object
-	 *
-	 * @param camera The camera
-	 * @param vpMatrix {@link Matrix4} The view-projection matrix
-	 * @param projMatrix {@link Matrix4} The projection matrix
-	 * @param vMatrix {@link Matrix4} The view matrix
-	 * @param parentMatrix {@link Matrix4} This object's parent matrix
-	 * @param pickerInfo The current color picker info. This is only used when an object is touched.
-	 */
+    /**
+     * Renders the object with a parent matrix, and optional color picking
+     *
+     * @param camera The camera
+     * @param vpMatrix {@link Matrix4} The view-projection matrix
+     * @param projMatrix {@link Matrix4} The projection matrix
+     * @param vMatrix {@link Matrix4} The view matrix
+     * @param parentMatrix {@link Matrix4} This object's parent matrix
+     * @param sceneMaterial The scene-wide Material to use, if any.
+     * @param isColorPicking True for color-picking render
+     */
 	public void render(Camera camera, final Matrix4 vpMatrix, final Matrix4 projMatrix, final Matrix4 vMatrix,
-			final Matrix4 parentMatrix, Material sceneMaterial) {
+			final Matrix4 parentMatrix, Material sceneMaterial, boolean isColorPicking) {
+
 		if (!mIsVisible && !mRenderChildrenAsBatch)
 			return;
 
 		Material material = sceneMaterial == null ? mMaterial : sceneMaterial;
-		preRender();
+
+        if (!isColorPicking) {
```

Comment:
I would like to see `preRender` always be called. In its current form, it validates that the VBOs have been created for the object. This could potentially be necessary for any object which is added after scene initialization but which was created without initializing its buffers (necessary to create objects on a thread other than the GL thread).